### PR TITLE
script: fetch and update all the zephyr modules from the STM32Cube

### PR DESCRIPTION
**Pull request by FRASTM**
For all the stm32 series:

1. give a script to fetch all the repo in $HOME/STM32Cube

2. give a script to update all the hal zephyr modules for all stm32 soc

Fetch repo from https://github.com/STMicroelectronics/STM32Cubethe
and checkout the HEAD, to prepare the upgrade of the zephyr modules
run with $ python fetch_cubes.py
Then
update zephyr modules with $ python update_cubes.py

Note that the scripts are for internal use on linux.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

File: scripts/update_cubes.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# This script will update the repo on each STM32Cube 
+# from the latest version on https://github.com/STMicroelectronics
+# 
+# usage : 'python update_cubes.py'
+
+import os
+import sys
+
+if not os.path.exists(os.getenv('ZEPHYR_BASE')):
+    print "Error: cannot find ./zephyr project"
```
Comment (gmarull) [inline scripts/update_cubes.py]: This code will run on Python 2 only (EOL)

File: scripts/update_cubes.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    sys.exit()
+
+module_path = os.path.join(os.getenv('ZEPHYR_BASE'), '../modules/hal/stm32/stm32cube')
+script_path = os.path.join(os.getenv('ZEPHYR_BASE'), '../modules/hal/stm32/scripts')
+os.chdir(script_path)
+
+for stmxx in os.listdir(module_path):
+ if os.path.isdir(os.path.join(module_path, stmxx)):
```
Comment (gmarull) [inline scripts/update_cubes.py]: Formatting does not follow PEP8 (I think it's good to follow it). `black` autoformatter may be useful.

File: scripts/update_cubes.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  print ("updating module " + stmxx[:-2])
+  os.system('python update_stm32_package.py -f ' + stmxx[:-2])
```
Comment (gmarull) [inline scripts/update_cubes.py]: I think `subprocess.Popen` is a better choice other than `os.system`.

File: scripts/fetch_cubes.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+# This script will update the repo on each STM32Cube 
+# from the latest version on https://github.com/STMicroelectronics
+# 
+# usage : 'python fetch_cubes.py'
+
+import os
+import sys
+import time
+
+cube_path = os.path.join(os.getenv('HOME'), 'STM32Cube_repo')
```
Comment (gmarull) [inline scripts/fetch_cubes.py]: That will only run on Linux, not sure if Windows is a target platform.

File: scripts/fetch_cubes.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+os.chdir(cube_path)
+
+for stm in os.listdir(cube_path):
+ if os.path.isdir(os.path.join(cube_path, stm)):
+  os.chdir(os.path.join(cube_path, stm))
+  print ("updating repo " + stm )
+  if os.system('git fetch origin') == 0:
+   os.system('git reset --hard origin/master')
+   os.system('git checkout master')
```
Comment (gmarull) [inline scripts/fetch_cubes.py]: Already said, `subprocess.Popen` may be preferred.

File: scripts/update_cubes.py (hunk 1 above)
Comment (FRASTM) [reply scripts/update_cubes.py:13]: ok

File: scripts/update_cubes.py (hunk 2 above)
Comment (FRASTM) [reply scripts/update_cubes.py]: formatted with black

File: scripts/update_stm32_package.py [hunk 6]
```diff
@@ -8,7 +8,9 @@
 # usage : 'python update_stm32_package.py <stm32_serie>'
 
 import os
-import sys; sys.dont_write_bytecode = True    # do not generate pyc file
+import sys
+
+sys.dont_write_bytecode = True  # do not generate pyc file
```
Comment (gmarull) [inline scripts/update_stm32_package.py:13]: Why is this needed?

File: scripts/update_stm32_package.py [hunk 7]
```diff
@@ -18,105 +20,113 @@
...
-arg_names = ['command', 'a1', 'a2', 'a3']
+arg_names = ["command", "a1", "a2", "a3"]
 args = map(None, arg_names, sys.argv)
 args = {k: v for (k, v) in args}
 
-if args['a1']:
-    if 'stm32' in args['a1']:
-        module_serie = args['a1']
-    elif '-c' in args['a1']:
+if args["a1"]:
```
Comment (gmarull) [inline scripts/update_stm32_package.py:29]: Use `argparse` module

File: scripts/update_stm32_package.py [hunk 8]
```diff
@@ -18,105 +20,113 @@
...
+    )
     while os.system(git_cmd) != 0:
         time.sleep(2)
     os.chdir(repo_path)
 else:
     # if already exists, then just clean and fetch
     os.chdir(repo_path)
-    os.system('git clean -fdx')
-    while os.system('git fetch') != 0:
+    os.system("git clean -fdx")
```
Comment (gmarull) [inline scripts/update_stm32_package.py:99]: `os.system` should be replaced by `subprocess.Popen`, it's not a safe choice.

File: scripts/update_stm32_package.py [hunk 9]
```diff
@@ -137,103 +147,114 @@
...
 os.system(git_cmd)
 
-git_cmd = 'git commit -am \"module' + previous_version + '\"'
+git_cmd = 'git commit -am "module' + previous_version + '"'
 print git_cmd
 os.system(git_cmd)
 
-git_cmd = 'git rebase --whitespace=fix HEAD~1'
+git_cmd = "git rebase --whitespace=fix HEAD~1"
 print "Remove remove trailing spaces: " + git_cmd
```
Comment (gmarull) [inline scripts/update_stm32_package.py:172]: Not Python 3 compatible.

File: scripts/readme.txt [hunk 10]
```diff
@@ -0,0 +1,81 @@
+-------------------------------------------------------
```
Comment (gmarull) [inline scripts/readme.txt:1]: Better use Markdown or Restructured text.

File: scripts/update_stm32_package.py (hunk 6 above)
Comment (FRASTM) [reply scripts/update_stm32_package.py:13]: It will merge only script files, not pyc ones

**Comment by erwango** [comment]
Superseded by https://github.com/zephyrproject-rtos/hal_stm32/pull/71
